### PR TITLE
Prepend http.root-path to permission paths without '/'

### PR DESCRIPTION
**Pull request by sberyozkin**
This PR relates to #25154,  it does what I did for 2.8.1 then reverted for 2.8.2, in a better non-breaking way as recommended by Erin.

I will follow up with another PR where I'll go through various OIDC paths carefully and see if the same logic can be applied.

### CODE REVIEW

**Comment by sberyozkin** [comment]
It should be safe for 2.9.0.CR1

File: docs/src/main/asciidoc/security-authorization.adoc [hunk 1]
```diff
@@ -177,6 +177,28 @@ quarkus.http.auth.permission.permit1.methods=GET,HEAD
...
+`quarkus.http.root-path` configuration property can be used to change an xref:http-reference.adoc#context-path[endpoint context path].
+
+If you'd like to have `quarkus.http.root-path` prepended automatically to the configured permission paths then do not use a forward slash, for example:
+
+[source,properties]
+----
+quarkus.http.auth.permission.permit1.paths=public/*,css/*,js/*,robots.txt
+----
+
+This configuration makes all the configured `quarkus.http.auth.permission.permit1.paths` relative to `quarkus.http.root-path` which is equal to `/` by default and in general will always end with `/`. Effectively this configuration is equivalent to
```
Comment (ebullient) [inline docs/src/main/asciidoc/security-authorization.adoc:191]:
```suggestion
This configuration is equivalent to the following: 
```

File: docs/src/main/asciidoc/security-authorization.adoc [hunk 2]
```diff
@@ -177,6 +177,28 @@ quarkus.http.auth.permission.permit1.methods=GET,HEAD
...
+
+[source,properties]
+----
+quarkus.http.auth.permission.permit1.paths=${quarkus.http.root-path}/public/*,${quarkus.http.root-path}/css/*,${quarkus.http.root-path}/js/*,${quarkus.http.root-path}/robots.txt
+----
+
+Clearly the configuration where a forward slash is omitted is much more compact and easier to read. Please avoid adding ${quarkus.http.root-path} directly to configured paths and drop the forward slash instead if you intend to prepend `quarkus.http.root-path`.
```
Comment (ebullient) [inline docs/src/main/asciidoc/security-authorization.adoc:198]:
```suggestion
A leading slash will change how the configured permission path is interpreted. The configured URL will be used as-is, and paths will not be adjusted if the value of `quarkus.http.root-path` is changed. For example: 

[source,properties]
----
quarkus.http.auth.permission.permit1.paths=/public/*,css/*,js/*,robots.txt
----

This configuration will only impact resources served from the fixed/static URL `/public`, which may not match your application resources if `quarkus.http.root-path` has been set to something other than `/`.
```

**Review by ebullient — COMMENTED** [review]
Writing is hard. See what you think.

**Review by ebullient — APPROVED** [review]

**Comment by sberyozkin** [comment]
Thanks, `quarkus-integration-test-jpa-oracle` has failed so far, it does not use these permission tests

**Comment by sberyozkin** [comment]
I think it is good to go

**Pull request merged**